File: Practica1Ruby/pkg/Practica1Ruby-0.0.1/lib/CarreraMontana.rb
require('./Carrera.rb');

public
class CarreraMontana < Carrera

	public
	def initialize(bicicletas)
		@bicicletas = bicicletas
    @numBicicletas = bicicletas.size
	end

  def clone()
    return CarreraMontana.new(@bicicletas)
	end
end

File: Practica1/src/practica1/CarreraMontana.java
package practica1;

import java.util.ArrayList;

public class CarreraMontana extends Carrera {
        
    public CarreraMontana(ArrayList <BicicletaMontana> listaBicicletas) {
        for (int i=0; i<listaBicicletas.size(); i++){
            this.bicicletas.add(listaBicicletas.get(i)); //Comprobas uppercasting
        }
        this.numBicicletas = listaBicicletas.size();
    }
    
}